Write valid leader without counting #527

TobiasNx opened the pull request:
Instead of just appending the incoming leader elements by `encode-marcmxl`, this adds the static values for Pos: 00-04, 10-16, 20-23.

By using `0`as default value for Pos: 00-04 and 12-16.

https://github.com/metafacture/metafacture-core/issues/527#issuecomment-2076585889

TobiasNx committed:
Write valid leader without counting #527

Instead of just appending the incoming leader elements, this adds the static values for Pos: 00-04, 10-16, 20-23.

By using `0`as default value for Pos: 00-04 and 12-16.

https://github.com/metafacture/metafacture-core/issues/527#issuecomment-2076585889

blackwinter commented:
What's the point of doing the "easy" solution now that we already have the "hard" solution?

TobiasNx commented:
The leader is always valid?

blackwinter reviewed (changes requested):
I guess I'm not too excited about this, but in the end I don't really care.

It should be tied to `ensureCorrectMarc21Xml="false"`, though.

Checkstyle violations should be fixed, tests should be added/fixed.

TobiasNx commented:
I will put this on hold, the failing test hints that it is not as easy: issue336_createRecordWithTopLevelLeader_defaultMarc21Xml

dr0i committed:
Add option 'validateLeader' (#567)

dr0i committed:
Merge #572 from branch '567-addEncodeMarc21ValidationAsOptional' of github.com:metafacture/metafacture-core

TobiasNx committed:
Merge remote-tracking branch 'origin/master' into 527-output-valid-leader

TobiasNx committed:
Add conditional that tests if the incoming leader has length of 8 #527

If the incoming leader is 8 Metafacture is now able to create correct leaders for marc xml with static default values.

TobiasNx commented:
> It should be tied to `ensureCorrectMarc21Xml="false"`, though.

@blackwinter how would I tie my adjustments to `ensureCorrectMarc21Xml="false"`,

blackwinter commented:
What's the actual use case here? When would this change apply in practice?

Because the implementation gets complicated by the fact that the setting for `ensureCorrectMarc21Xml` needs to be propagated down into the `encoder` instance.

TobiasNx commented:
Always create valid marc xml records from marc binary  besides the option to create correct calculate records.
When helping Verbundgruppe with Metafacture at least in my opinion it becomes more handy to output always a valid `leader`.

blackwinter commented:
So binary MARC always has a leader of length 8, is that what you're saying? Or only in their case?
